src/data/set/basic.lean

```diff
@@ -1817,6 +1820,24 @@ lemma range_diff_image {f : α → β} (H : injective f) (s : set α) :
 subset.antisymm (range_diff_image_subset f s) $ λ y ⟨x, hx, hy⟩, hy ▸
   ⟨mem_range_self _, λ ⟨x', hx', eq⟩, hx $ H eq ▸ hx'⟩
 
+/-- We can use the axiom of choice to pick a preimage for every element of `range f`. -/
+noncomputable def range_splitting (f : α → β) : range f → α := λ x, x.2.some
+
+lemma apply_range_splitting (f : α → β) (x : range f) : f (range_splitting f x) = x :=
+x.2.some_spec
+
+attribute [irreducible] range_splitting
+
+@[simp] lemma comp_range_splitting (f : α → β) : f ∘ range_splitting f = coe :=
+by { ext, simp only [function.comp_app], apply apply_range_splitting, }
+
+lemma left_inverse_range_splitting (f : α → β) :
```

Comment:
Let's add a note here to see `equiv.of_injective` for this result as an equiv when `f` is injective.

On that note, does it make sense to change equiv.of_injective to use these functions?

Reply:
Added a note. I don't think it matters to write one function in terms of the other, as they are all `noncomputable` anyway, and have no good definitional properties.